### DISCUSSION TITLE
Latency/RTT optimization

### DISCUSSION THREAD

**Discussion by mrouili** [srsran/srsRAN_Project#147] [General Help] [answered]
Hello srsRAN community, I was wondering if there is a way to improve the achieved latency using the srsRAN 5G SA gNB (monolithic). Without any changes to the provided example configurations, I am able to get around 20 ms. Are there any parameters or changes that could be made to achieve a lower and more consistent latency ? Thank you. 

My testbed information:

- Operating System: Ubuntu 22.04 LTS
- CPU: i9-10980XE CPU @ 3.5 GHz
- RAM: 32 GB
- USRP X310
- Google Pixel 7 PRO
- Band 78 TDD

**Comment by ismagom** [comment]
I guess you are measuring RTT with ping, no? In these kind of scenarios the SR latency is the biggest impact. Try to reduce it to the maximum. Sometimes generating some small but continuous uplink traffic helps since the UE has an uplink grant every slot.
The TDD pattern also has an impact

**Reply by mrouili** [reply]
I see, thank you for your reply. So i'm guessing that in terms of parameters (gNB/core) and configuration files, there is no parameters or values that I could change to improve this right ?

**Reply by frankist** [reply]
We will soon expose the SR period as a configuration parameter of the gnb application. In the current release it is hard coded but can be changed.

**Reply by frankist** [reply]
Hi @mrouili,

With the new public release, you can try shorter SR periods. Just be aware of matching the SR period to the TDD pattern period. This should allow you to achieve latencies below 20msec. Just be aware that, for very short TDD patterns (e.g. <2.5msec), it might be hard to fit the CSI-RS, SSB and SIB1. We are currently working on improving this.

**Reply by mrouili** [reply]
@frankist Thank you ! I will report on this once I try it out.

**Reply by mrouili** [reply]
Hello again @frankist , I finally got the opportunity to test it out. However, I can't figure out how to include the sr period parameter into the gnb config file, from digging into the code , I can see that the exact name of the variable is sr_period_ms if i'm not wrong, passing it as a flag (--sr_period_ms) when deploying the gnb didn't work. Under what block of the gnb config file should I add the parameter (cell_cfg didn't work), thanks !

**Reply by pgawlowicz** [reply, marked as answer]
it should be added in gnb config file at:
```
cell_cfg:
...
  pucch:
    sr_period_ms: 10
...
```
